pkg/services/secrets/kvstore/cache.go

```diff
@@ -0,0 +1,69 @@
+package kvstore
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type decryptionCache struct {
+	decryptionCache map[string]*cachedDecrypted
+	sync.Mutex
+}
+
+type cachedDecrypted struct {
+	lastAccess time.Time
+	updated    time.Time
+	value      string
+}
+
+const (
+	recentTimeout = time.Second * 5
+)
+
+func newDecryptionCache() decryptionCache {
+	return decryptionCache{
+		decryptionCache: make(map[string]*cachedDecrypted),
+	}
+}
+
+func (c *decryptionCache) key(orgId int64, namespace string, typ string) string {
```

Comment:
Nit. Can potentially be simple function without receiver on `decryptionCache`. At least `func (c decryptionCache)` since it doesn't change anything on `decryptionCache`.